src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -163,6 +164,23 @@ export default function connectHierarchicalMenu(renderFn) {
           false
         );
       },
+
+      dispose(helper) {
+        // unmount widget from DOM
+        unmountFn();
+
+        // compute nextState for the search
+        const nextState = helper
+          .getState()
+          .removeHierarchicalFacetRefinement(hierarchicalFacetName)
+          .removeHierarchicalFacet(hierarchicalFacetName);
+
+        if (nextState.maxValuesPerFacet === limit) {
+          nextState.maxValuesPerFacet = undefined;
```

Comment:
In the spirit of the usage of the SearchParameters, you should use setQueryParameter and not mutate the parameter directly. If you want to avoid yet another copy of the state, you can use `setQueryParameters` which let you pass a complete object.